input/parser.go

```diff
@@ -2,68 +2,80 @@ package input
 
 import (
 	"fmt"
-	"strings"
+	"github.com/ekanite/ekanite/parser"
+	"log"
 )
 
-var (
-	fmtsByStandard = []string{"rfc5424"}
-	fmtsByName     = []string{"syslog"}
+const (
+	Rfc5424Standard   = "RFC5424"
+	Rfc5424Name       = "syslog"
+	WatchguardFirebox = "Watchguard"
+	WatchguardName    = "M200"
 )
 
+type StatsCollector func(key string, delta int64)
+
+type LogParser interface {
+	Parse(raw []byte, result *map[string]interface{})
+	CompileMatcher()
```

Comment:
This function name seems a little specific for an interface definition. Presumably not all future parsers would have regex inside -- so this exposes implementation assumptions in the interface. Perhaps this should be called "Init()" or something similar?

Reply:
Great insight and feedback - hadn't considered a non-regex impl.